sycl/doc/extensions/SPIRV/SPV_INTEL_spec_conditional.asciidoc

```diff
@@ -0,0 +1,694 @@
+cases, I think that `G(i)` and `G(c)` will both be so simple that it will be
+easy to prove whether one implies the other.)  How should the spec handle this?
+Are the validation rules _requirements_ that a validation tool must perform, or
+are they just rules about what SPIR-V is not valid?  If they are not
+requirements, then we could leave the wording as I have it and then each
+validation tool would be free to implement the validation checks as thoroughly
+as it wants.
+
+
+== New tokens defined by this extension
+
+=== New capabilities
+
+The module must declare that it uses the following capability in order to use
+any of the decorations or instructions defined in the sections below.  If
+declared, this capability must be unconditionally declared via *OpCapability*.
+It may not be conditionally declared via *OpConditionalCapabilityINTEL*.
+
+[cols="1,15,5",options="header",width="100%"]
+|===
+2+^| Capability | Implicitly Declares
+| ????
+| *SpecConditionalINTEL* +
+Module conditionally enables code based on the value of a specialization
+constant.
+|
+|===
+
+=== New decorations
+
+[cols="1,10,5,5",options="header",width="100%"]
+|====
+2+^| Decoration | Extra Operands | Enabling Capabilities
+| ????
+| *ConditionalINTEL* +
+May be applied only to *OpFunction*, global (module scope) *OpVariable*, type
+declarations (*OpTypeXXX*), or constant instructions (*OpConstantXXX* or
+*OpSpecConstantXXX*).  Indicates that the decorated instruction must be removed
+if the value of the specialization constant identified by _Condition_ is
+*false*.  The _Condition_ must be a _Boolean type_ scalar.
+
+If the decorated instruction is *OpFunction*, the function and all of the
+instructions it contains are removed when the specialization constant is
+*false*.  If the function has an associated *OpEntryPoint*, that is also
+removed.
+| _<id>_ _Condition_
+|*SpecConditionalINTEL*
+|====
+
+=== New instructions
+
+[cols="1,1,2*3",width="100%"]
+|===
+3+|*OpConditionalCapabilityINTEL* +
+ +
+Declare a capability that is conditionally used by this module, depending on
+the value of a specialization constant.
+
+The _Capability_ is used by this module only if the specialization constant
+identified by _Condition_ is *true*.  The _Condition_ must be a _Boolean type_
+scalar.
+
+1+|Capability: +
+*SpecConditionalINTEL*
+| 3
+| ????
+| _<id>_ _Condition_
+| _Capability_
+|===
+
+[cols="1,1,4*3",width="100%"]
+|===
+5+|*OpBranchSpecConstantINTEL* +
+ +
+If the specialization constant _Condition_ is *true*, branch to _True Label_,
+otherwise branch to _False Label_.  The _Condition_ must be a _Boolean type_
+scalar.
+
+The consecutive blocks from _True Label_ to _True End_ (inclusive) are called
+the "then" range of this instruction.  If the _Condition_ is *false*, this
+range of blocks is removed from the module.  The module need not declare any
+capabilities used by these instructions if they are removed.
```

Comment:
```suggestion
capabilities used exclusively by these instructions if they are removed.
```

Reply:
Resolving this because we decided to go with an alternate design that does not require a SPIR-V extension.